presto-spi/src/main/java/io/prestosql/spi/type/SqlTimestamp.java

```diff
@@ -43,7 +45,26 @@ public static SqlTimestamp fromMillis(int precision, long millis)
 
     public static SqlTimestamp newInstance(int precision, long epochMicros, int picosOfMicro)
     {
-        return newInstanceWithRounding(precision, epochMicros, picosOfMicro);
+        if (precision <= 6) {
+            if (picosOfMicro != 0) {
+                throw new IllegalArgumentException("Expected picosOfMicro to be 0 for precision <= 6: " + picosOfMicro);
```

Comment:
Mentioning epochMicros and precision as part of the exc msg may help tracing down the problem, should the exception occur anywhere outside of our tests.

introducing `private void checkArgument(...)` may help increase readability.